### ISSUE TITLE
Renaming the discrimnant of a union should always rename all occurrences

### ISSUE THREAD

**Issue by RyanCavanaugh** [Microsoft/TypeScript#21288]
**TypeScript Version:**  master

**Search Terms:** rename all locations find references discriminated union

**Code**

```ts
type Result =
    { kind: "success", code: number } | // 1
    { kind: "failure", message: string }; // 2
function fn(): Result {
    if (Math.random() > 0.5) {
        return { kind: "failure", message: "oops" } // 3
    } else {
        return { kind: "success", code: 200 } // 4
    }
}

const r = fn();
if (r.kind === "failure") { // 5
    r.message;
} else if (r.kind === "success") { // 6
    r.code;
}
```

There are 6 occurrences of `kind` in this program (annotated above); renaming at locations 1 and 2 does not produce an error-free program, locations 3 and 4 do, location 5 does, but location 6 does not.

**Playground Link:** [link](http://www.typescriptlang.org/play/#src=type%20Result%20%3D%0D%0A%20%20%20%20%7B%20kind%3A%20%22success%22%2C%20code%3A%20number%20%7D%20%7C%0D%0A%20%20%20%20%7B%20kind%3A%20%22failure%22%2C%20message%3A%20string%20%7D%3B%0D%0Afunction%20fn()%3A%20Result%20%7B%0D%0A%20%20%20%20if%20(Math.random()%20%3E%200.5)%20%7B%0D%0A%20%20%20%20%20%20%20%20return%20%7B%20kind%3A%20%22failure%22%2C%20message%3A%20%22oops%22%20%7D%0D%0A%20%20%20%20%7D%20else%20%7B%0D%0A%20%20%20%20%20%20%20%20return%20%7B%20kind%3A%20%22success%22%2C%20code%3A%20200%20%7D%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0Aconst%20r%20%3D%20fn()%3B%0D%0Aif%20(r.kind%20%3D%3D%3D%20%22failure%22)%20%7B%0D%0A%20%20%20%20r.message%3B%0D%0A%7D%20else%20if%20(r.kind%20%3D%3D%3D%20%22success%22)%20%7B%0D%0A%20%20%20%20r.code%3B%0D%0A%7D%0D%0A)

**Related Issues:** ?
